shared/tree-sitter-extractor/src/extractor/mod.rs

```diff
@@ -15,11 +22,29 @@ pub mod simple;
 /// `RUST_LOG` and `CODEQL_VERBOSITY` (prioritized in that order),
 /// falling back to `warn` if neither is set.
 pub fn set_tracing_level(language: &str) {
-    tracing_subscriber::fmt()
+    tracing_subscriber::registry()
+        .with(default_subscriber_with_level(language))
+        .init();
+}
+
+/// Create a `Subscriber` configured with the tracing level based on the environment variables
+/// `RUST_LOG` and `CODEQL_VERBOSITY` (prioritized in that order), falling back to `warn` if neither is set.
+pub fn default_subscriber_with_level(
+    language: &str,
+) -> Filtered<
+    tracing_subscriber::fmt::Layer<
+        tracing_subscriber::Registry,
+        DefaultFields,
+        Format<tracing_subscriber::fmt::format::Full, ()>,
+    >,
+    EnvFilter,
+    tracing_subscriber::Registry,
+> {
```

Comment:
hmm, this bypasses our own current numerical env we use for this `CODEQL_RUST_EXTRACTOR_VERBOSE` or `CODEQL_RUST_EXTRACTOR_OPTION_VERBOSE`, which I would prefer to keep (AFAIK `CODEQL_VERBOSITY` is not a generic codeql cli setting, so actually I think the env extractor namespacing should be used, although admittedly verbosity is still not enrolled in our `codeql-extractor.yml` file).